### ISSUE TITLE
Balancing Manacosts at OAA levels. 

### ISSUE THREAD

**Issue by POCKETMONEY99** [OpenAngelArena/oaa#2936]
Manacosts are not in a good state at the moment. Some heroes are heavily impacted by the huge manacost spikes at OAA levels. Heroes like tidehunter are better off not skilling their ultimate at OAA levels as they cost most of the heroes mana pool. There is also the nasty suprise of having an item build which allows you to sustain your mana costs to leveling up and not having enough mana to cast your spells. **Some heroes would be saved from having 300 mana cost basic spells if their dota version had a static mana cost.** 

Other heroes dont care about having 2k mana cost ultimates and 400 mana cost basic spells. Heroes like tinker still have near infinite mana pools despite having huge mana cost spikes at OAA levels. 

Currently mana costs on some heroes is implemented in a way which "makes problems then sells solutions" in an attempt to make mana and mana regen items relevant. 

My proposal:
- Linearise the mana cost scaling for all spells. Just make the spells mana cost scale by the same amout it was scaling at dota levels. Spells like nether ward, and sources of mana break may need values adjusting after this change.  
- Nerf mana cost reduction on kaya/bloodstone. Most heroes wont care about this but outliers like storm and tinker will. (Tinker has near infinite mana at the moment) Bloodstone manaloss/manacost reduction from "12 15 18 21 24" to "12 14 16 18 20" Kaya Variants mana cost reduction from "16 18 22 28 36" to "18 21 24 27 30". 
- Reduce the amount of excess mana regen available from items. We dont need items like aether lens giving +24 mana regen at maximum level. (Tinker abuses this item well) Additionally with the reduced mana cost the need for huge mana regen will be unessecary. 

Example: Tidehunter. Tidehunter gains (1.7 x 20 x 12)= 408 Extra Mana from levels 30-50
Gush: "90 100 110 120 160 220" to "90 100 110 120 130 140"
Anchor Smash: "30 40 50 60 110 160" to "30 40 50 60 70 80"
Ravage: "150 225 325 425 525" (Some spells like Ravage have strange mana cost scalings. I just used the largest increase in mana cost per dota levels and used that scaling for OAA levels.

**Comment by DarkoniusXNG** [comment]
I disagree on some points. 

> Linearise the mana cost scaling for all spells.

1) All INT heroes should have high mana costs and their mana costs should not scale linearly. There are some few exceptions ofc but those are rare.

> Heroes like tidehunter are better off not skilling their ultimate at OAA levels as they cost most of the heroes mana pool. There is also the nasty suprise of having an item build which allows you to sustain your mana costs to leveling up and not having enough mana to cast your spells. 

2. Heroes that have mana issues are STR and AGI heroes that dont buy a single item that gives them mana in some way. In other words greedy carries have mana issues. Some heroes do have weird and unjustified mana costs but scaling them linearly is not always a good option.

> Nerf mana cost reduction on kaya/bloodstone. 

3) Manacost/manaloss reduction could use a rescale, but now its not the time for that.

> Global mana regen nerf.

4) Global mana regen nerf wouldn't work. You would also need to nerf all +mana items and all +int items which is **very bad**. Aether Lens is the only mana regen item that stands out and could use a nerf in all aspects. But people always cry when something is nerfed.

5) Tidehunter is a bad example. Ravage and Gush mana costs are completely fine. Ofc he would have mana issues if he goes full rightclick build. Anchor Smash mana cost stands out because it was meant to discourage braindead Tidehunter carry.

Usual scaling of mana costs at OAA levels are: max lvl mana cost is either 2x, 3x or 4x of last dota level or +100. **Depends if the spell is spammable, how is the hero played and which items the hero usually buy.** If all mana costs scaled linearly at OAA levels, game would become more like League of Legends, item builds would be braindead and mana management would matter even less. Imagine Tidehunter with 5 Daedaluses always having mana to use Ravage. Also linearizing mana costs would make CM aura and KOTL Chakra Magic even more useless.

**Comment by POCKETMONEY99** [comment]
>     1. All INT heroes should have high mana costs and their mana costs should not scale linearly. There are some few exceptions ofc but those are rare.

Why do INT heroes need extreme mana costs just because they are intelligence heroes. When you look at dota heroes, simillar spell archetypes tend to have simillar mana costs. Most mobility spells hover around 60 mana cost, most single target stuns are around 140 mana etc. Just because a hero is intelligence doesnt mea Some item builds on intelligence heroes like right click lina or QoP cannot function because all their spells just multiply in mana costs. 

For example Wraith King Wraith Blast has the mana cost "95 110 125 140 150 160" while Ogre Magi Fireblast has the mana cost "75 85 95 105 210 315" Both of these spells are fairly simillar yet Ogres Triples in mana cost. 

In dota mana costs on most heroes is a way to restrain **early game** harrass and limit peoples farming speed. Your not supposed to have mana issues on most heroes once you are level 30 and 6 slotted. If a hero is naturally mana hunrgy I will still buy mana items. 

In OAA this is semi true but some heroes regain mana issues once they are 6 slotted for some arbitary reason. This is a nasty suprise and is just bad. When I am playing a hero like Mars or Faceless Void my build can be completely fine for my mana issues at dota levels but when I get into OAA levels I can regain mana issues. At this point I have already decided which items I am using, repurchasing items isnt very easy. Heavily mana dependent heroes like Storm and Tinker are the exception not the norm for dota heroes. 
> 
> > Heroes like tidehunter are better off not skilling their ultimate at OAA levels as they cost most of the heroes mana pool. There is also the nasty suprise of having an item build which allows you to sustain your mana costs to leveling up and not having enough mana to cast your spells.
> 
>     1. Heroes that have mana issues are STR and AGI heroes that dont buy a single item that gives them mana in some way. In other words greedy carries have mana issues. Some heroes do have weird and unjustified mana costs but scaling them linearly is not always a good option.

Even after this change "greedy" carries will still have mana issues. (Not buying a mana item isn't really greedy but w/e). 
>     1. Manacost/manaloss reduction could use a rescale, but now its not the time for that.

Why is now not the time for that? Also why do kaya variants have nerfed level 1 values on mana loss reduction. This can be reverted now that bloodstone outclasses these items. 

> > Global mana regen nerf.
> 
>     1. Global mana regen nerf wouldn't work. You would also need to nerf all +mana items and all +int items which is **very bad**. Aether Lens is the only mana regen item that stands out and could use a nerf in all aspects. But people always cry when something is nerfed.

I dont see why it wouldnt work. It would only be relatively minor mana regen changes. Most heroes wouldn't care, its mainly so outliers are kept in line. All of the items will still be just as relevant as they were previously. Mana regen is pretty much in excess after 1 item on most heroes. 

>     2. Tidehunter is a bad example. Ravage and Gush mana costs are completely fine. Ofc he would have mana issues if he goes full rightclick build. Anchor Smash mana cost stands out because it was meant to discourage braindead Tidehunter carry.

Ravage mana costs are terrible. When I have completed my item build if I have solved my mana issues with my level 1 items, they shoudn't reappear at a later level for no real reason. 

> Usual scaling of mana costs at OAA levels are: max lvl mana cost is either 2x, 3x or 4x of last dota level or +100. **Depends if the spell is spammable, how is the hero played and which items the hero usually buy.** If all mana costs scaled linearly at OAA levels, game would become more like League of Legends, item builds would be braindead and mana management would matter even less. Imagine Tidehunter with 5 Daedaluses always having mana to use Ravage. Also linearizing mana costs would make CM aura and KOTL Chakra Magic even more useless.

This game wont turn into league of legends with the spell spam due to reduced mana costs. Cooldown is the main limiting factor for most heroes. Its mainly the omni present 40% CDR combined with most basic skills having sub 10 second default cooldowns. Mana costs arent supposed to be an issue for 99% of heroes once they have a few items. In OAA everyone gets 6 slotted at 10 minutes which allows them to solve any mana issues they would have. Any hero with mana issues will still have to purchase mana items. Increasing mana costs heavily at OAA levels just make problems for no reason. 

The main goal of this change is to make it so if you solve your mana issues with your first 6 items with your dota mana costs it stays constant through OAA levels. Currently that isnt the case on a fair number of heroes. The secondary goal is to limit the amount of excess mana regen you get so storm and tinker are easier to balance. 

KOTL and CM aura will still be nice for the people who dont get mana items. KOTL still has the CDR component for his spell. CM aura just needs buffing heavily at dota levels then its good. Carries will still appriciate the extra mana.

**Comment by Larenall** [comment]
This makes so little sense to me

> Nerf mana cost reduction on kaya/bloodstone. Most heroes wont care about this but outliers like storm and tinker will. (Tinker has near infinite mana at the moment)

Idk why u mention tinker and storm soo often as if they are OP heroes and require a nerf when in reallity its all other int heroes that need the nerf cuz its them who have infinite mana. I have no clue why u think tinker has infinity mana, he did have a lot of effective mana(but no infinite) b4 darkonious nerfed rearm, thats true, but its no longer the case. Id say rn tinker has a well fitting amount of effective mana and it isnt what has to be changed about the hero. While tinker finally is a good hero, storm isnt one, he needs a buff, not a nerf.

**Comment by POCKETMONEY99** [comment]
> 
> 
> This makes so little sense to me
> 
> > Nerf mana cost reduction on kaya/bloodstone. Most heroes wont care about this but outliers like storm and tinker will. (Tinker has near infinite mana at the moment)
> 
> Idk why u mention tinker and storm soo often as if they are OP heroes and require a nerf when in reallity its all other int heroes that need the nerf cuz its them who have infinite mana. I have no clue why u think tinker has infinity mana, he did have a lot of effective mana(but no infinite) b4 darkonious nerfed rearm, thats true, but its no longer the case. Id say rn tinker has a well fitting amount of effective mana and it isnt what has to be changed about the hero. While tinker finally is a good hero, storm isnt one, he needs a buff, not a nerf.

I mention storm and tinker because these are the 2 heroes which are limited by their mana rather then cooldowns. They are instead limited by the amount of mana they have. My proposed changes is meant to be a quality of life change, along with it being a net buff for most heroes, while limiting the amount mana items can compound. If I put these changes into the game, people are not going to need huge mana regen, because their spells dont have exponential mana cost scaling at OAA levels. The mana cost reduction nerf is realtively minor. 

 But the main goals of this patch is to give everyone more simillar mana costs, rather then it being INT heroes get huge mana costs for arbitary reasons, along with some other heroes which usually dont need to get mana items. The second goal is to reduce the amount of excess mana regen available. On most heroes the excess mana regen is unecessary and the regen would be extra unessecary with my reduced mana costs for all heroes

**Comment by DarkoniusXNG** [comment]
Am I the only one that sees flaws and contradictions in your posts?

> Why do INT heroes need extreme mana costs just because they are intelligence heroes...

Did you read what I wrote about mana regen nerf. If you linearise mana costs, you need to nerf **every +mana regen item, every +mana item and every +int item**. Ok? By doing that you would make most int heroes more dead than they are. You would kill Skywrath though :). But if you dont nerf every item like I said, INT heroes will literally have infinite mana. Only antimage would be able to deal with them.

> When I am playing a hero like Mars or Faceless Void my build can be completely fine for my mana issues at dota levels but when I get into OAA levels I can regain mana issues. 

Mana issues are a balance thing and not just because of things you mentioned (early game harass and farming speed) . If most of the STR heroes didnt have mana issues, STR heroes would be the best heroes to pick every game. For example those 2 heroes (Mars and Void) always have mana issues unless you go more than 2 mana items (even in normal dota). 

> In OAA this is semi true but some heroes regain mana issues once they are 6 slotted for some arbitrary reason. This is a nasty surprise and is just bad. 

That's your opinion. Also let me quote you from one other issue. ''I dont think infinite mana is an issue on 99% of heroes. **Heroes that have mana issues will have mana issues throughout the game if they dont buy mana items. This is fine.**'' - your words. Emphasis on THIS IS FINE. 

> Cooldown is the main limiting factor for most heroes. 

Do heroes have mana issues or not? Most of the rant looks like mana is the main limiting factor and now I see this sentence.
Also cooldowns are becoming a more meaningless factor each day you make a pull request that reduces cooldown on some ability (pudge hook) or ultimate. 

> Mana costs arent supposed to be an issue for 99% of heroes once they have a few items.

I never had mana issues when I play support. Carries also should not have mana issues when they are near supports because supports have Ward Stack mana regen aura or Vladmir mana regen aura.
INT heroes also never have mana issues (ignore Storm and Tinker for now) unless they are playing against mana burn or Pugna ward.

> The main goal of this change is to make it so if you solve your mana issues with your first 6 items with your dota mana costs it stays constant through OAA levels. Currently that isnt the case on a fair number of heroes.

Name those heroes, and I will see which mana costs need change. But I am 100000% sure that changing mana costs for ALL heroes would be terrible. It would be a disaster. For example I allowed linearising Luna mana costs because the hero is bad and because it made sense.

> If mana regen is more reasonable, storm can have his ultimate nerfs reduced/reverted so he can actually play the game. The auto remnant talent can also be readded. On revaluation the auto remnant talent wasn't the issue, it was the huge excess of mana regen allowing storm to play that playstyle.

No, no and no. Mana regen was an issue on Storm on multiple ocassions (when Refresher gave insane mana regen; when Bloodstones stacked completely etc.) but that was not the main reason for Storm nerfs. Auto remnant talent WAS the main issue. Auto remnant talent was nerfed (and now removed and it will stay removed) because you CANT INTERACT with that kind of playstyle (ask Chris and Max for more explanations). You can say just use bkb lol. 
Main reasons for mana cost nerf on Storm ult are:
1) excess manacost/manaloss reduction (this is also an answer to question why is lvl 1 kaya variant nerfed).
2) because map is smaller than dota map
3) because duel arenas are small. In other words it was to prevent infinite zipping in duel arenas. 

> Some item builds on intelligence heroes like right click lina or QoP cannot function because all their spells just multiply in mana costs.

If you go rightclick QoP or Lina, why would you ever use spells if your rightclicks do more dmg? Also if you dont buy INT items on rightclick QoP or Lina thats your mistake, not because mana costs are too high.

**Comment by TheUnholyWrath** [comment]
If a spell exists that you literally can't cast unless you buy a mana/stat item then that would be a problem (the only hero i've found that fits is lion [make finger cost 2343 at max level or I riot]). An actual "problem" under all this squabbling is spells with manacosts that don't scale in dota were never given oaa levels. 

People would be bitching and moaning just as much about agi heroes having mana problems if half (roughly) of the spells didn't scale past a level 1 dota value. (slark is completely broken nerf him his spells don't scale in manacost wah wah wah).

Manacosts almost certainly shouldn't be linear because of how xp works, if they are thats likely from buffs to weak heroes in the past

Some of the manacosts on int heroes would almost certainly exist for 2 reasons: either the spell has a bullshit cooldown (lion) or refresher shenanigans (warlock). 

Spark Wraith does 1400 damage on a 2.5 second cooldown for 80 mana. BuT iF hE wErE aN iNt HeRo...

there are obviously some outliers, ravage for example really doesn't seem worth the manacost/increase. why not just try give them a lovetap and see if it makes a difference (or just buy a skadi). 

back in my day when every single item in the game gave 60 all stats we didn't have no damn mana problems... 

In conclusion, if you want mana to be an issue in the early game fix the double spawn, if you don't want manacosts to be a problem in the late game, either change the values, standardise/raise int gain or stop valuing a 7th daedalus so much higher than a skadi (or a hex).

**Comment by POCKETMONEY99** [comment]
I have reworded my initial post. Hopefully you understand what I am trying to do. Personally I dont see any flaws or contradictions in my reasoning. 

> Did you read what I wrote about mana regen nerf. If you linearise mana costs, you need to nerf **every +mana regen item, every +mana item and every +int item**. Ok? By doing that you would make most int heroes more dead than they are. You would kill Skywrath though :). But if you dont nerf every item like I said, INT heroes will literally have infinite mana. Only antimage would be able to deal with them.

I was only planning on reducing mana regen at OAA levels, along with a rescaling of the manaloss reduction items. I am not intending to change intelligence or raw mana values on items. This change is also intended to only affect the "OAA" section of the game not Dota. I am not planning on heavily nerfing mana items. I am just reducing the amount of **excess mana regen** available, so heroes balanced around mana regen can't abuse certain items. 

> Mana issues are a balance thing and not just because of things you mentioned (early game harass and farming speed) . If most of the STR heroes didnt have mana issues, STR heroes would be the best heroes to pick every game. For example those 2 heroes (Mars and Void) always have mana issues unless you go more than 2 mana items (even in normal dota).

STR heroes having mana issues arise because generally heroes have high stat gain for their primary stat and lower stat gain for their other stat. Therefore most strength heroes have low intelligence. Then most spells in dota are balanced around their qualities rather then what the hero is casting the spell. This is currently my issue in OAA, because spell mana costs are heavily determined by which hero is casting it, rather then what the spell does. If you take a spell in dota and look at other similar spells they tend to have simillar mana costs. 

> > In OAA this is semi true but some heroes regain mana issues once they are 6 slotted for some arbitrary reason. This is a nasty surprise and is just bad.

> That's your opinion. Also let me quote you from one other issue. ''I dont think infinite mana is an issue on 99% of heroes. **Heroes that have mana issues will have mana issues throughout the game if they dont buy mana items. This is fine.**'' - your words. Emphasis on THIS IS FINE.

This statement is true. After my changes it will still be true. If you dont buy a single mana item on heroes with poor int gain and high mana cost spells, you will still run out of mana. It will just take longer. Most heroes will get infinite mana from 6 level 1 items. Some heroes only need 1 item for inifinite mana. Its all relative to the hero. I am reducing the amount of mana regen on items etc simply because you dont need it. All it does is allow certain heroes to abuse it. 

OAA isnt just one teamfight then you can go back to base. You need mana for casting spells on bosses, farming camps etc. I am not removing mana costs, I am making them more reasonable, along with "harmonising" them across the board, rather then giving specific heroes high mana costs just because they usually buy bloodstone etc. 

> > Cooldown is the main limiting factor for most heroes.
> Do heroes have mana issues or not? Most of the rant looks like mana is the main limiting factor and now I see this sentence.

Mana and cooldown limit spell casting. Cooldown is the main limiting factor on alot heroes once they are 6 slotted. Mana dependency will remain for certain heroes if they dont opt to purchase mana regeneration. 


> I never had mana issues when I play support. Carries also should not have mana issues when they are near supports because supports have Ward Stack mana regen aura or Vladmir mana regen aura.
> INT heroes also never have mana issues (ignore Storm and Tinker for now) unless they are playing against mana burn or Pugna ward.

I am making mana costs reasonable for all heroes at OAA levels. The heroes which need this change most are the ones who mana pools will barely increase from leveling up etc. This change only benefits INT heroes, even if it is only minor. 
 
> > The main goal of this change is to make it so if you solve your mana issues with your first 6 items with your dota mana costs it stays constant through OAA levels. Currently that isnt the case on a fair number of heroes.

> Name those heroes, and I will see which mana costs need change. But I am 100000% sure that changing mana costs for ALL heroes would be terrible. It would be a disaster. For example I allowed linearising Luna mana costs because the hero is bad and because it made sense.

I don't see how it would be terrible. Some caster heroes have very modest mana cost and they arent out of control op. The only suspect heroes are tinker and storm spirit. But I would look at thoes heroes personally. This is the intention of reducting the amount of excess mana regen available after most heroes obtain "infinite mana regen". 

There is no point in listing heroes because I think all heroes need it. 


> If you go rightclick QoP or Lina, why would you ever use spells if your rightclicks do more dmg? Also if you dont buy INT items on rightclick QoP or Lina thats your mistake, not because mana costs are too high.

You still buy mana items on these heroes, and you still need to cast your spells. However you can't stack 6 mana items like if you were playing caster.
